Implement OAuth 2.0 token introspection

centromere opened the pull request:
This PR provides a bare-bones implementation of [RFC 7662](https://datatracker.ietf.org/doc/html/rfc7662). Example usage:

```rust
#[derive(Debug, Deserialize, Serialize)]
struct Roles {
    roles: Vec<String>,
}

#[derive(Debug, Deserialize, Serialize)]
struct CustomKeycloakFields {
    realm_access: Roles,
    resource_access: HashMap<String, Roles>,
    name: String,
    email: String,
}

impl CompactJson for CustomKeycloakFields {}

async fn validator(req: ServiceRequest, credentials: BearerAuth) -> Result<ServiceRequest, (actix_web::Error, ServiceRequest)> {
    // ...
    let token = Token::from(openid::Bearer {
        access_token: String::from(credentials.token()),
        token_type: String::from("Bearer"),
        scope: None,
        state: None,
        refresh_token: None,
        expires_in: None,
        id_token: None,
        extra: None,
    });

    let introspection: TokenIntrospection<CustomKeycloakFields> = openid_client.request_token_introspection(&token).await.unwrap();
    if introspection.active {
        // ...
    } else {
        // ...
    }
}
```

kilork commented:
@centromere Hi, this looks really cool. What is the meaning of "Begin" here, do you mean there are more functions to cover from RFC? Could you add to README part about the RFC and support status (which features are implemented), if it is applicable?

centromere commented:
@kilork, Hi. The coverage seems complete to me. I wrote "Begin" because I anticipated needing to make changes to better conform to the project structure. I've added a note to the README and updated the commit.

kilork commented:
@centromere I did small corrections and going to merge it to master. The crate itself is a mix of two other crates, it is difficult to say, if project structure well defined or is it a result of evolution :)
